src/PrestaShopBundle/Command/ConfigCommand.php

```diff
@@ -96,12 +95,10 @@ class ConfigCommand extends Command
     private $idLang;
 
     public function __construct(
-        LegacyContextLoader $legacyContextLoader,
         ShopConfigurationInterface $configuration,
         LanguageDataProvider $languageDataProvider
     ) {
         parent::__construct();
-        $legacyContextLoader->loadGenericContext();
```

Comment:
I don't know if this component is useful indeed, but if it was added I guess there was a reason Maybe @tswfi could tell us more about it?

@atomiix  I thought alternatively maybe `$legacyContextLoader->loadGenericContext()` could be called in the `execute` command instead? this way it's still in use and it should not be called in the constructor as soon as the service is handled by the container

Reply:
By the way, it's weird that the constructor is always called as soon as the service is registered I thought services were only built on demand
Maybe we could define them as lazy as well to avoid this (https://symfony.com/doc/current/service_container/lazy_services.html)

Reply:
Cant remember why it was required but I think there was something that was needed from the context by DebugMode..
But if it works correctly without it then I guess it can be removed.

Might even be that it was required for translations which were later removed and context was left behind.

Reply:
> I thought alternatively maybe $legacyContextLoader->loadGenericContext() could be called in the execute command instead? 

@jolelievre that's effectively an option I was thinking of but I also wanted to see if it was really required at all and it didn't seem to so I just removed it 😄

>Maybe we could define them as lazy as well to avoid this

This would definitely solve the issue too

Reply:
OK seems there is no use at all then removing it is the best choice indeed